nautobot/core/api/metadata.py

```diff
@@ -57,3 +62,184 @@ def get_field_info(self, field):
             ]
             field_info["choices"].sort(key=lambda item: item["display"])
         return field_info
+
+
+class NautobotProcessingMixin(ProcessingMixin):
+    """Processing mixin  to account for custom field types and behaviors for Nautobot."""
+
+    def _get_type_map_value(self, field: SerializerType):
+        """Overload default to add "required" as a default mapping."""
+        # This adds "required" as a default type mapping compared to drf_react_template core.
+        result = {
+            "type": field.style.get("schema:type"),
+            "enum": field.style.get("schema:enum"),
+            "format": field.style.get("schema:format"),
+            "widget": field.style.get("ui:widget"),
+            "required": field.style.get("schema:required"),
+            "readOnly": field.style.get("schema:readOnly"),
+        }
+        result_default = self.TYPE_MAP.get(type(field).__name__, {})
+        for k in result_default:
+            if result[k] is None:
+                result[k] = result_default[k]
+        return result
+
+
+class NautobotSchemaProcessor(NautobotProcessingMixin, SchemaProcessor):
+    """
+    SchemaProcessor to account for custom field types and behaviors for Nautobot.
+    """
+
+    def _get_field_properties(self, field: SerializerType, name: str) -> Dict[str, Any]:
```

Comment:
The challenge is adding support for `ManyRelatedField` for nested child relationships. Let's revisit next week, because I've had to make a handful of other changes to adhere properly to JSON schema and we should discuss strategy here.